tony-history-server/app/controllers/JobConfigPageController.java

```diff
@@ -32,16 +35,21 @@ public JobConfigPageController(Config config) {
   public Result index(String jobId) {
     HdfsConfiguration conf = Configuration.getHdfsConf();
     FileSystem myFs = HdfsUtils.getFileSystem(conf);
+    Cache<String, List<JobConfig>> cache = CacheWrapper.getConfigCache();
 
     if (myFs == null) {
       return internalServerError("Failed to initialize file system");
     }
 
-    List<JobConfig> listOfConfigs = new ArrayList<>();
+    List<JobConfig> listOfConfigs;
     String tonyHistoryFolder = config.getString("tony.historyFolder");
     Path xmlPath = new Path(tonyHistoryFolder + jobId + "/config.xml");
-
+    if (cache.asMap().containsKey(jobId)) {
+      listOfConfigs = cache.getIfPresent(jobId);
```

Comment:
I think you only need one call to the cache here. You can do `JobConfig jobConfig = cache.getIfPresent(jobId)`. Then check if `jobConfig` is null or not.